src/openrct2/ride/Ride.cpp

```diff
@@ -1132,13 +1132,11 @@ void ride_remove_peeps(Ride* ride)
 
             if (exitDirection == 255)
             {
-                int32_t x = peep->next_x + 16;
-                int32_t y = peep->next_y + 16;
-                int32_t z = peep->next_z * 8;
+                CoordsXYZ newLoc = { peep->NextLoc.ToTileCentre(), peep->NextLoc.z };
                 if (peep->GetNextIsSloped())
-                    z += 8;
-                z++;
-                sprite_move(x, y, z, peep);
+                    newLoc.z += 8;
```

Comment:
```suggestion
                    newLoc.z += COORDS_Z_STEP;
```

Reply:
That would be incorrect. 8 is already in large coordinates.